src/mscorlib/src/System/Random.cs

```diff
@@ -114,11 +121,39 @@ public Random()
           return retVal;
       }
 
+
+      [ThreadStatic]
+      private static Random t_threadRandom;
+      private static Random s_globalRandom = new Random(GenerateGlobalSeed());
+
+      /*=====================================GenerateSeed=====================================
+      **Returns: An integer that can be used as seed values for consecutively
+                 creating lots of instances on the same thread within a short period of time.
+      ========================================================================================*/
+      private static int GenerateSeed() {
+          if (t_threadRandom == null) {
+              int seed;
+              lock (s_globalRandom) {
+                  seed = s_globalRandom.Next();
+              }
+              t_threadRandom = new Random(seed);
+          }
+          return t_threadRandom.Next();
+      }
+
+      /*==================================GenerateGlobalSeed====================================
+      **Action:  Creates a number to use as global seed.
+      **Returns: An integer that is safe to use as seed values for thread-local seed generators.
+      ==========================================================================================*/
+      private static int GenerateGlobalSeed() {
+          return Guid.NewGuid().GetHashCode();
```

Comment:
The short answer is 'yes' I do worry :smile:

However, in this case, I think using a Guid should be fine. I would still prefer that we don't introduce a dependency to Guid, but there doesn't seem to be any other way of introducing process specific entropy (which would solve the problem of multiple processes starting withing 15ms).

Regarding Desktop -- the code does not automatically make its way into Desktop. We have a [process in place] to track those kinds of changes. (https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/triage.md#porting-to-net-framework).